ntoskrnl/mm/ARM3/virtual.c

```diff
@@ -756,32 +723,17 @@ MiDeleteVirtualAddresses(IN ULONG_PTR Va,
 
                     /* Use this to detect address gaps */
                     PointerPte++;
+
+                    PrototypePte++;
```

Comment:
Note that the `PointerPte++` above was introduced in commit https://github.com/reactos/reactos/commit/f606b36cffeaa3ab78bb730db2fcee80ba61d2f6 in order to make the code below:
```c
        /* Check if we exited the loop regularly */
        AddressGap = (PointerPte != MiAddressToPte(Va));
```
do this detection.

My question here: Is this new `PrototypePte++` unrelated to the increment of PointerPte?
If so I would suggest you move it certainly above (just after the `/* Continue with the next PDE */` probably)

Reply:
`PointerPte++` and `PrototypePte++` update are normally done at the end of the loop here:
[https://github.com/reactos/reactos/blob/8ba6c6689070842ea2c03e7d20f4d9e7f177f479/ntoskrnl/mm/ARM3/virtual.c#L766](https://github.com/reactos/reactos/blob/8ba6c6689070842ea2c03e7d20f4d9e7f177f479/ntoskrnl/mm/ARM3/virtual.c#L766)

Reply:
I think it's unrelated but I would like to keep the logic the same as the end of the loop (Update Va -> Update PointerPte -> Update PrototypePte) for readability.
[https://github.com/reactos/reactos/blob/8ba6c6689070842ea2c03e7d20f4d9e7f177f479/ntoskrnl/mm/ARM3/virtual.c#L764](https://github.com/reactos/reactos/blob/8ba6c6689070842ea2c03e7d20f4d9e7f177f479/ntoskrnl/mm/ARM3/virtual.c#L764)